### PR TITLE
Update album takestamps

### LINKED ISSUE #738: [Enhancement] Sort order of albums: Ignore images without timestamps
Hello,

I sort my albums by `max_takestamp DESC`. Unfortunately I have a few albums with pictures that have no timestamp. These albums are always displayed at the first position. It would be nice if the images without timestamps were ignored in the sorting order.

### PR DESCRIPTION
**Pull request by kamil4**
Fixes #738.

We did not update the album min/max takestamps after a successful extraction.

I ended up rewriting most of its logic since it made no sense to me...

### CODE REVIEW

**Review by ildyria — APPROVED** [review]
LGTM (not tested)

File: app/Console/Commands/Takedate.php [hunk 1]
```diff
@@ -73,13 +73,14 @@ public function handle()
 			$url = Storage::path('big/' . $photo->url);
 			if (file_exists($url)) {
 				$info = $this->metadataExtractor->extract($url, $photo->type);
-				if ($photo->takestamp == '') {
+				if ($info['takestamp'] != null) {
 					$photo->takestamp = $info['takestamp'];
-				}
-				if ($photo->save()) {
-					$this->line($i . ': Takestamp updated for ' . $photo->title);
+					if ($photo->save()) {
```
Comment (d7415) [inline app/Console/Commands/Takedate.php:70]: Do we need handling if this fails?

File: app/Console/Commands/Takedate.php [hunk 2]
```diff
@@ -73,13 +73,14 @@ public function handle()
...
+						$this->line($i . ': Takestamp updated for ' . $photo->title);
+					}
+					AlbumUpdate::update_takestamps($photo->album, [$photo->takestamp], true);
```
Comment (d7415) [inline app/Console/Commands/Takedate.php]: Would this be better before the `}`, so that it only updates if the photo update was successful?

**Review by d7415 — COMMENTED** [review]
Seems to make sense. A couple of queries.

**Comment by kamil4** [comment]
Thanks for your comments, @d7415. They inspired me to have another critical look at this part of the code and prompted a more thorough rewrite of not just `Takedate.php` but also `ExifLens.php` and `VideoData.php`. All three now feature a more sensible SQL query and updated diagnostics which makes it clear if something was actually updated or not.

@ildyria, @d7415, the latest commit significantly expands the scope of this PR so would you mind making another review round?

File: app/Console/Commands/VideoData.php [hunk 3]
```diff
@@ -98,41 +97,47 @@ public function handle()
...
 					$photo->focal = $info['focal'];
+					$updated = true;
 				}
-				if ($info['aperture'] !== '') {
+				if ($photo->aperture == '' && $info['aperture'] !== '') {
 					$photo->aperture = $info['aperture'];
+					$updated = true;
 				}
-				if ($info['takestamp'] !== null) {
-					$photo->takestamp = $info['takestamp'];
```
Comment (d7415) [inline app/Console/Commands/VideoData.php:115]: Presumably this is handled elsewhere so was redundant?

File: app/Console/Commands/VideoData.php [hunk 4]
```diff
@@ -67,7 +66,7 @@ public function handle()
 			)
 		);
 
-		$photos = Photo::where('type', 'like', 'video/%')
+		$photos = Photo::whereIn('type', $this->photoFunctions->getValidVideoTypes())
```
Comment (d7415) [inline app/Console/Commands/VideoData.php:69]: Less permissive, but IIRC more in line with how we handle this elsewhere, so makes sense.

**Review by d7415 — APPROVED** [review]
Makes sense to me (untested, as usual)

File: app/Console/Commands/VideoData.php (hunk 3 above)
Comment (kamil4) [reply app/Console/Commands/VideoData.php:115]: We have a separate `lychee:takedate` command specifically for that. Removing it from here makes it more consistent with `lychee:exif_lens` which doesn't do it.

File: app/Console/Commands/VideoData.php (hunk 4 above)
Comment (kamil4) [reply app/Console/Commands/VideoData.php:69]: Actually, check: https://github.com/LycheeOrg/Lychee/blob/2b0664a09dea2ca9533c1879bffad7c97b36c519/app/ModelFunctions/PhotoFunctions.php#L51-L62
As you can see, one video type (MPG) is sometimes recognized as `image/x-tga`, so this change actually makes it more inclusive...

**Review by ildyria — APPROVED** [review]
LGTM

**Review by ildyria — APPROVED** [review]

**Pull request merged**